File: bit_plane.m
A=imread('C:\Users\skconan\Desktop\Workspace\digital_image_processing\images\bill.jpg');
A = rgb2gray(A);
imshow(A);
B=bitget(A,1);
figure,
subplot(4,2,1);imshow(logical(B));title('Bit plane 1');

B=bitget(A,2);
subplot(4,2,2);imshow(logical(B));title('Bit plane 2');

B=bitget(A,3);
subplot(4,2,3);imshow(logical(B));title('Bit plane 3');


B=bitget(A,4);
subplot(4,2,4);imshow(logical(B));title('Bit plane 4');

B=bitget(A,5);
subplot(4,2,5);imshow(logical(B));title('Bit plane 5');

B=bitget(A,6);
subplot(4,2,6);imshow(logical(B));title('Bit plane 6');

B=bitget(A,7);
subplot(4,2,7);imshow(logical(B));title('Bit plane 7');


B=bitget(A,8);
subplot(4,2,8);imshow(logical(B));title('Bit plane 8');

File: bit_plane.py
'''
    File name       : bit_plane.py
    Author          : skconan
    Date created    : 2018/08/15
'''

import cv2 as cv
import numpy as np
import constant as CONST
import  matplotlib.pyplot as plt

def dec2bin(dec,index):
    return int(bin(dec).replace('0b','').zfill(8)[index])

def bit_plane():
    gray = cv.imread(CONST.IMG_PATH+'/bill.jpg',0)
    # gray = cv.resize(gray,(0, 0), fx=0.5, fy=0.5)
    r,c = gray.shape
    cv.imshow('original',gray.copy())
    _, axarr1 = plt.subplots(8)
    # tmp = np.zeros((r,c),np.uint8)
    tmp = 0
    
    res = []
    # plane 1 least significant 
    for k in range(8):
        res.append([])
        for i in range(r):
            res[k].append([])
            for j in range(c):
                res[k][i].append(dec2bin(gray[i][j],7-k)*255)
        if k > 0:
            res[k] = np.array(res[k],np.uint8)
        else:
            res[k] = np.array(res[k],np.uint8)

        cv.imshow('result '+str(k+1),res[k])
        cv.waitKey(-1)

if __name__ == '__main__':
    bit_plane()
